testcases/simple_query.js

```diff
@@ -716,6 +720,69 @@ if (typeof(tests) !== "object") {
+            });
+        }
+    }
+
+    /**
+     * Set of test cases which stress the performance of projections with dotted paths as well
+     * as wide projections.
+     */
+    const singlePathThreeComponentDocGenerator = function(i) {
+            return {a: {b: {c: i, d: i}}}
+    };
+
+    addInclusionExclusionTestCase("ProjectionDottedField.SinglePathThreeComponents", singlePathThreeComponentDocGenerator, {"a.b.c": 1, _id: 0}, {"a.b.d": 0, _id: 1});
+
+    const singlePathThreePathComponentsDeepProjectionDocGenerator = function(i) {
+        return {a: [
+                {b: [{c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}]},
+                {b: [{c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}]},
+                {b: [{c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}]},
+                {b: [{c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}]},
+                {b: [{c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}, {c: i, d: i}]}]};
+    };
+
+    addInclusionExclusionTestCase("ProjectionDottedField.SinglePathThreeComponentsNestedArrays", singlePathThreePathComponentsDeepProjectionDocGenerator, {"a.b.c": 1, _id: 0}, {"a.b.d": 0, _id: 1});
+
+    const singlePathSixPathComponentsDeepProjectionDocGenerator = function(i) {
+        const arrayElement = {b: {c: [
+                    {d: {e: [{f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}]}},
+                    {d: {e: [{f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}]}},
+                    {d: {e: [{f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}]}},
+                    {d: {e: [{f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}]}},
+                    {d: {e: [{f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}, {f: i, g: i}]}}]}};
+        return {a: [arrayElement, arrayElement, arrayElement, arrayElement, arrayElement]};
+    };
+
+    addInclusionExclusionTestCase("ProjectionDottedField.SinglePathSixComponentsNestedArrays", singlePathSixPathComponentsDeepProjectionDocGenerator,  {"a.b.c.d.e.f": 1, _id: 0}, {"a.b.c.d.e.g": 0, _id: 1});
+
+    const topLevelWideProjectionDocGenerator = function(i) {
+        return {a: i, b: i, c: i, d: i, e: i, f: i, g: i, h: i, i: i, j: i};
+    }
+
+    addInclusionExclusionTestCase("WideProjectionTopLevelField", topLevelWideProjectionDocGenerator, {_id: 0, b: 1, c: 1, d: 1, e: 1, f: 1, g: 1, h: 1, i: 1, j: 1}, {_id: 1, b: 0, c: 0, d: 0, e: 0, f: 0, g: 0, h: 0, i: 0, j: 0});
+
+    const nestedWideProjectionDocGenerator = function(i) {
+        return {n: topLevelWideProjectionDocGenerator(i)};
+    }
+
+    addInclusionExclusionTestCase("WideProjectionNestedField", nestedWideProjectionDocGenerator, {_id: 0, n: {b: 1, c: 1, d: 1, e: 1, f: 1, g: 1, h: 1, i: 1, j: 1}}, {_id: 1, n: {b: 0, c: 0, d: 0, e: 0, f: 0, g: 0, h: 0, i: 0, j: 0}});
```

Comment:
is this line too long?
```suggestion
    addInclusionExclusionTestCase("WideProjectionNestedField" /* name */, 
                                                           nestedWideProjectionDocGenerator, 
                                                           {_id: 0, n: {b: 1, c: 1, d: 1, e: 1, f: 1, g: 1, h: 1, i: 1, j: 1}} /* inclusionSpec */, 
                                                           {_id: 1, n: {b: 0, c: 0, d: 0, e: 0, f: 0, g: 0, h: 0, i: 0, j: 0}} /* exclusionSpec */);
```